tests/phpunit/tests/blocks/block.php

```diff
@@ -413,4 +413,120 @@ function test_passes_content_to_render_callback() {
 		$this->assertSame( 'abc', $block->render() );
 	}
 
+	/**
+	 * Tests `construct_wp_query_args` that is
+	 * helper function that constructs a WP_Query args
+	 * array from a `Query` block properties.
+	 *
+	 * @since 5.8.0
+	 */
+	public function test_construct_wp_query_args() {
+		$this->registry->register(
+			'core/example',
+			array( 'uses_context' => array( 'query' ) )
+		);
+
+		$parsed_blocks    = parse_blocks( '<!-- wp:example {"ok":true} -->a<!-- wp:example /-->b<!-- /wp:example -->' );
+		$parsed_block     = $parsed_blocks[0];
+		$block_no_context = new WP_Block( $parsed_block, array(), $this->registry );
+
+		// It should return the default context when block doesn't use `query` context.
+		$query = construct_wp_query_args( $block_no_context, 1 );
```

Comment:
Should I create different test functions here or is it okay with one that contains some different scenarios? @gziolo

Reply:
It's better to have one test per assertion and your comments are good indicators of what should be the best way to split the logic.